qiskit/tools/_compiler.py

```diff
@@ -47,6 +48,15 @@ def compile(circuits, backend,
                       'Please pass an empty PassManager() instance instead',
                       DeprecationWarning)
 
+    if shots > backend.configuration().max_shots:
```

Comment:
what happens when you using a state vector simulator.

Reply:
good point. I defined `max_shots` for the statevector simulators to be equal to 1. 

Therefore requesting more shots will raise an exception here.